MatrixSDK/Crypto/Verification/Data/MXQRCodeDataCoder.m

```diff
@@ -0,0 +1,296 @@
+/*
+ Copyright 2020 The Matrix.org Foundation C.I.C
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MXQRCodeDataCoder.h"
+#import "MXQRCodeDataCodable.h"
+#import "MXQRCodeData.h"
+#import "MXQRCodeDataBuilder.h"
+#import "MXBase64Tools.h"
+
+#import "MXVerifyingAnotherUserQRCodeData.h"
+
+static NSString* const kQRCodeFormatPrefix = @"MATRIX";
+static NSUInteger const kQRCodeFormatVersion = 2;
+
+@interface MXQRCodeDataCoder()
+
+@property (nonatomic, strong) NSData *prefixData;
+@property (nonatomic) NSUInteger supportedQRCodeVersion;
+@property (nonatomic, strong) MXQRCodeDataBuilder *qrCodeDataBuilder;
+
+@end
+
+@implementation MXQRCodeDataCoder
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self)
+    {
+        self.prefixData = [kQRCodeFormatPrefix dataUsingEncoding:NSASCIIStringEncoding];
+        self.supportedQRCodeVersion = kQRCodeFormatVersion;
+        self.qrCodeDataBuilder = [MXQRCodeDataBuilder new];
+    }
+    return self;
+}
+
+- (MXQRCodeData*)decode:(NSData*)data
+{
+    NSInteger dataByteCount = data.length;
+    
+    NSInteger minimumDataByteCount = 10;
+    
+    if (dataByteCount < minimumDataByteCount)
+    {
+        NSLog(@"[MXQRCodeDataCoder] Data byte count is too short");
+        return nil;
+    }
+    
+    NSInteger totalReadBytesCount = 0;
+    NSInteger currenReadBytesCount = 0;
+    
+    NSInputStream *inputStream = [NSInputStream inputStreamWithData:data];
+    [inputStream open];
+    
+    // Found the ASCII string "MATRIX"
+    
+    NSUInteger prefixLength = self.prefixData.length;
+    uint8_t prefixBuffer[prefixLength];
+    
+    currenReadBytesCount = [inputStream read:prefixBuffer maxLength:prefixLength];
+    
+    if (currenReadBytesCount < 0)
+    {
+        NSLog(@"inputStream.streamError: %@", inputStream.streamError);
+        return nil;
+    }
+    
+    totalReadBytesCount += currenReadBytesCount;
+    
+    NSData *foundPrefixData = [[NSData alloc] initWithBytes:prefixBuffer length:prefixLength];
+    
+    if (![self.prefixData isEqualToData:foundPrefixData])
+    {
+        NSLog(@"[MXQRCodeDataCoder] decode: Invalid prefix");        
+        return nil;
+    }
+    
+    // Check the QR code version
+    
```

Comment:
I am not confortable with theses comments between `\n`. I do not feel they separate things and make these sections more readable

Perhaps add one more `\n` before and remove the one after:
```suggestion

```suggestion
    
    
    // Check the QR code version 
```